crypto/Pos/Crypto/Signing/Types/Signing.hs

```diff
@@ -140,7 +140,9 @@ instance B.Buildable (ProxyCert w) where
 
 -- | Convenient wrapper for secret key, that's basically ω plus
 -- certificate.
-data ProxySecretKey w = ProxySecretKey
+--
+-- The invariant is that the certificate is valid.
```

Comment:
What is "valid"? You should add "with respect to `validateProxySecretKey`" I think.